src/api_proxy/service_control/request_builder.cc

```diff
@@ -383,7 +383,9 @@ Status set_credential_id(const SupportedLabel& l, const ReportRequestInfo& info,
     ASSERT(!info.api_key.empty(),
            "API Key must be set, otherwise consumer would not be verified.");
     std::string credential_id("apikey:");
-    credential_id += info.api_key;
+    credential_id = credential_id
+           + (info.check_response_info.api_key_uid.empty() ? info.api_key
+                      : info.check_response_info.api_key_uid);
```

Comment:
NIT: define credential_id as constchar above and use absl::StrCat to concat string.

Reply:
done.

Reply:
I mean the constexpr char like in l348

Reply:
done.